examples/ecs/one_shot_systems.rs

```diff
@@ -0,0 +1,73 @@
+//! Demonstrates the use of "one-shot systems", which run once when triggered
+//!
+//! These can be useful to help structure your logic in a push-based fashion,
+//! reducing the overhead of running extremely rarely run systems
+//! and improving schedule flexibility.
+//!
+//! See the [`SystemRegistry`](bevy::ecs::SystemRegistry) docs for more details.
+
+use bevy::ecs::system::Callback;
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .add_startup_system(count_entities)
+        .add_startup_system(setup)
+        // One shot systems are interchangeable with ordinarily scheduled systems.
+        // Change detection, Local and NonSend all work as expected
+        .add_system_to_stage(CoreStage::PostUpdate, count_entities)
+        // Registered systems can be called dynamically by their label
+        // These must be registered in advance, or commands.run_system will panic
+        // as no matching system was found
+        .register_system(button_pressed)
+        .register_system(slider_toggled)
+        // One-shot systems can be used to build complex abstractions
+        // to match the needs of your design.
+        // Here, we model a very simple component-linked callback architecture.
+        .add_system(evaluate_callbacks)
+        .run();
+}
+
+// Any ordinary system can be run via commands.run_system or world.run_system
+//
+// Chained systems, exclusive systems and systems which themselves run systems cannot be called in this way.
+fn count_entities(all_entities: Query<()>) {
+    dbg!(all_entities.iter().count());
+}
+
+#[derive(Component)]
+struct Triggered;
+
+fn setup(mut commands: Commands) {
+    commands
+        .spawn()
+        // The Callback component is defined in bevy_ecs,
+        // but wrapping this (or making your own customized variant) is easy.
+        // Just stored a boxed SystemLabel!
+        .insert(Callback::new(button_pressed))
+        .insert(Triggered);
+    // This entity does not have a Triggered component, so its callback won't run
+    commands.spawn().insert(Callback::new(slider_toggled));
+    commands.run_system(count_entities);
+}
+
+fn button_pressed() {
+    println!("A button was pressed!");
+}
+
+fn slider_toggled() {
+    println!("A slider was toggled!");
+}
+
```

Comment:
I found this confusing the first time I read through the example, I though this evaluation system was something special, and callbacks had to be `Component`s with triggers.

I think this could be improved by keeping the example much simpler - like the unit tests you wrote with a simple counter. I finally "got it" after seeing the unit test. This might be a great way to show the on-demand usage of `Callback`s, without the risk of the confusion I experienced. :slightly_smiling_face: 

May I suggest something like this?
```suggestion
/// A resource we will modify with a callback
struct Counter(usize)

/// A bevy system that we will run from a callback instead of inserting it in the schedule
fn increment_counter(counter: ResMut<Counter>) {
    *counter = counter + 1;
}

fn setup(commands: mut Commands) {
    // Callbacks provide a way to run a system as if it was a function call with full access to the ECS.
    commands.insert_resource(Counter(0));
    // Instead of inserting the system into the schedule, we'll invoke it manually using a callback.
    let increment_callback = Callback::new(increment_counter);
    // Recall that commands are queued up and executed at the *end* of the stage.
    commands.run_callback(increment_callback);
}
```